### ISSUE TITLE
L2 memory mapping duplication

### ISSUE THREAD

**Issue by kangoojim** [pulp-platform/pulpissimo#178]
Hi,

I stumbled over the following lines in soc_interconnect.sv from the pulp_soc repo:

```
    always_comb
    begin
        FC_DATA_add_int = FC_DATA_add_i;
        if(FC_DATA_add_i[31:20] == 12'h000)
            FC_DATA_add_int[31:20] = 12'h1C0;
    end
```

https://github.com/pulp-platform/pulp_soc/blob/04d0bbc526582c0b38d5096bf283753fca339d29/rtl/pulp_soc/soc_interconnect.sv#L397-L402

Effectively, this maps the L2 Memory twice into the address range of the core's/FC's data port. So the memory locations 32'h1C0... are also accessible through the addresses 32'h000...

I am not sure whether the lines implement any vital functionality. But this mapping is not mentioned in the datasheet in the memory map section. So it might be worth updating either this HW component or the datasheet.

The issue could also be considered a security (access control) violation, since it enables circumvention of a possible memory protection mechanisms (either in SW or by PMP).

**Comment by bluewww** [comment]
Yes this mirroring is intentional. It allows you to access the first 2KiB of 1C00'000 to be accessed with a single lw/sw skipping the otherwise necessary LUI. In the pulp compiler there is the "tiny" attribute that allows you to place data there.

Whether this is a good idea is debateable, especially since there is the gp register that allows you to do a similar thing.
